Model/ResourceModel/Products.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Klaviyo\Reclaim\Model\ResourceModel;
+
+use Klaviyo\Reclaim\Helper\Logger;
+
+class Products extends \Magento\Framework\Model\ResourceModel\Db\AbstractDb
+{
+    protected $_klaviyoLogger;
+    protected function _construct()
+    {
+        $this->_init('kl_products', 'id');
+    }
+
+    public function __construct(
+        \Magento\Framework\Model\ResourceModel\Db\Context $context,
+        Logger $klaviyoLogger
+    )
+    {
+        parent::__construct($context);
+        $this->_klaviyoLogger = $klaviyoLogger;
+    }
+
+    public function updateRowsToMoved($ids)
+    {
+        if (empty($ids)) {
+            return;
+        }
+
+        $this->getConnection()->update(
+            $this->getMainTable(),
+            ['status' => 'MOVED'],
+            ['id IN(?)' => $ids]
+        );
+    }
+
+    public function deleteMovedRows($ids)
+    {
+        if (empty($ids)) {
+            return;
+        }
+
+        $this->getConnection()->delete(
+            $this->getMainTable(),
+            ['id IN(?)' => $ids]
+        );
+    }
+}
```

Comment:
Looks like this logic is duplicated in the events model as well. Could be beneficial to create an abstract class for these methods especially if we intend to add other resource models in the future e.g. orders.